R/approx-cdf.R

```diff
@@ -144,11 +145,16 @@ approx_cdf_from_quantiles <- function(quantiles, probs) {
   assert_that(!is.unsorted(quantiles))
   assert_that(is_double(probs))
   assert_that(all(0 <= probs & probs <= 1))
-  ##
+
+  # NOTE: What's the purpose of this run-length encoding?
   rle.quantiles <- rle(quantiles)
+  # NOTE: cdf.xs appears to be identical to quantiles.
```

Comment:
```suggestion
```

Above ties case is where this can be different.  Ditto for below.  I've just suggested removing these comments but explanatory comments that consider/mention the ties case would be nice as well.

Sorry there's no explanation here, but there is [here](https://docs.google.com/presentation/d/1oOvhecbGNYyacVzJdWN0NFVHXAtFRvAfn3-IBcAc5A0/edit#slide=id.g218340d910f_0_0).  Unfortunately the names for all the vars here are also bad.

We want to represent a piecewise linear + jumps function.  There are various ways you might think of representing this.  One way to do this would be represent the piecewise linear part and the jump part separately.  Another way, used here, is to keep a list of x values, F(x-) values (cdf.ys.before), and F(x) values (cdf.ys.at).  (I think both of these representation options have their awkward moments when using, but not sure which has fewer.)